engine/access/ingestion/tx_error_messages/tx_error_messages_requester.go

```diff
@@ -20,41 +20,49 @@ type TransactionErrorMessagesRequesterConfig struct {
 }
 
 // TransactionErrorMessagesRequester is a thin wrapper over a TxErrorMessagesCore that manages the retrieval
-// of transaction error messages with retry and backoff configurations.
+// of transaction error messages with retry and backoff configurations for a specific execution result.
 type TransactionErrorMessagesRequester struct {
-	core   *TxErrorMessagesCore
-	config *TransactionErrorMessagesRequesterConfig
+	core            *TxErrorMessagesCore
+	config          *TransactionErrorMessagesRequesterConfig
+	executionResult *flow.ExecutionResult
 }
 
 func NewTransactionErrorMessagesRequester(
 	core *TxErrorMessagesCore,
 	config *TransactionErrorMessagesRequesterConfig,
+	executionResult *flow.ExecutionResult,
 ) *TransactionErrorMessagesRequester {
 	return &TransactionErrorMessagesRequester{
-		core:   core,
-		config: config,
+		core:            core,
+		config:          config,
+		executionResult: executionResult,
 	}
 }
 
-func (r *TransactionErrorMessagesRequester) RequestTransactionErrorMessagesByBlockID(
+// RequestTransactionErrorMessages fetches transaction error messages for the specific
+// execution result this requester was configured with.
+func (r *TransactionErrorMessagesRequester) RequestTransactionErrorMessages(
 	ctx irrecoverable.SignalerContext,
```

Comment:
does this need to be an `irrecoverable.SignalerContext` or could it be a regular context?